select unix_timestamp() gets an ambiguous error message

## Bug Report

Please answer these questions before submitting your issue. Thanks!

### 1. Minimal reproduce step (Required)

<!-- a step by step guide for reproducing the bug. -->

```mysql
select unix_timestamp();
```

### 2. What did you expect to see? (Required)

No error log is shown.

### 3. What did you see instead (Required)

Got an ambiguous error, `[ERROR] [terror.go:375] ["encountered error"] [error="[types:1265]Data truncated for column '%s' at row %d"]`

### 4. What is your TiDB version? (Required)

<!-- Paste the output of SELECT tidb_version() -->
master

--- a/expression/builtin_time.go
+++ b/expression/builtin_time.go
@@ -4760,7 +4760,9 @@ func (b *builtinUnixTimestampCurrentSig) evalInt(row chunk.Row) (int64, bool, er
 		return 0, true, err
 	}
 	intVal, err := dec.ToInt()
-	terror.Log(err)
+	if !terror.ErrorEqual(err, types.ErrTruncated) {
+		terror.Log(err)
+	}
 	return intVal, false, nil
 }
 
@@ -4800,7 +4802,9 @@ func (b *builtinUnixTimestampIntSig) evalIntWithCtx(ctx sessionctx.Context, row 
 		return 0, true, err
 	}
 	intVal, err := dec.ToInt()
-	terror.Log(err)
+	if !terror.ErrorEqual(err, types.ErrTruncated) {
+		terror.Log(err)
+	}
 	return intVal, false, nil
 }
 

--- a/expression/builtin_time_vec.go
+++ b/expression/builtin_time_vec.go
@@ -464,7 +464,9 @@ func (b *builtinUnixTimestampCurrentSig) vecEvalInt(input *chunk.Chunk, result *
 		return err
 	}
 	intVal, err := dec.ToInt()
-	terror.Log(err)
+	if !terror.ErrorEqual(err, types.ErrTruncated) {
+		terror.Log(err)
+	}
 	n := input.NumRows()
 	result.ResizeInt64(n, false)
 	intRes := result.Int64s()
@@ -2271,7 +2273,9 @@ func (b *builtinUnixTimestampIntSig) vecEvalInt(input *chunk.Chunk, result *chun
 				return err
 			}
 			intVal, err := dec.ToInt()
-			terror.Log(err)
+			if !terror.ErrorEqual(err, types.ErrTruncated) {
+				terror.Log(err)
+			}
 			i64s[i] = intVal
 		}
 	}

